### PR TITLE
Archive flagship v3 core

### PR DESCRIPTION
**Pull request by ryley-o**
## Description of the change

Archive Flagship V3 core in favor of deriving from Engine for improved modularity.

Note that some tests will still use V3 Flagship, some intentional (e.g. active contracts that need to remain compatible with Flagship V3 core as-deployed), but some might be unintentional, and should be removed as they are found.

Some tests are updated to use V3 Engine instead of V3 Flagship.

### CODE REVIEW

**Comment by ryley-o** [comment]
unintentionally merged 😓 would appreciate a retroactive review on this 🙏

File: packages/contracts/test/core/V3/prohibition/GenArt721CoreV3_Integration_PROHIBITION.test.ts [hunk 1]
```diff
@@ -377,7 +377,7 @@ for (const coreContractName of coreContractsToTest) {
           expect(coreType).to.be.equal("GenArt721CoreV3_Engine_Flex");
         } else {
           // coreType is same for GenArt721CoreV3 & GenArt721CoreV3_Explorations
-          expect(coreType).to.be.equal("GenArt721CoreV3");
+          throw new Error("Untested core contract version");
```
Comment (lindsgil) [inline packages/contracts/test/core/V3/prohibition/GenArt721CoreV3_Integration_PROHIBITION.test.ts:380]: nit - should we remove the comment on line 369 - 370 (GH won't let me comment up there 😓 )

**Comment by lindsgil** [comment]
LGTM - just one comment RE: a comment 🙏

File: packages/contracts/test/core/V3/prohibition/GenArt721CoreV3_Integration_PROHIBITION.test.ts (hunk 1 above)
Comment (ryley-o) [reply packages/contracts/test/core/V3/prohibition/GenArt721CoreV3_Integration_PROHIBITION.test.ts:380]: thanks - updated on feature branch directly in [e7b73d8](https://github.com/ArtBlocks/artblocks-contracts/pull/1470/commits/e7b73d854f61a0b415c144a7d9233a98c77fc4a0)

**Pull request merged**